packages/vscode-extension/src/project/project.ts

```diff
@@ -182,26 +149,55 @@ export class Project
 
   private setupAppRoot() {
     const newAppRoot = findAndSetupNewAppRootFolder();
+
     const oldApplicationContext = this.applicationContext;
     this.applicationContext = new ApplicationContext(newAppRoot);
     oldApplicationContext.dispose();
 
-    this.reload("reboot");
+    const oldDeviceSessionsManager = this.deviceSessionsManager;
+    this.deviceSessionsManager = this.deviceSessionsManager = new DeviceSessionsManager(
```

Comment:
I think one assignment should be sufficient
```suggestion
    this.deviceSessionsManager = new DeviceSessionsManager(
```